File: median.rb
# http://rosettacode.org/wiki/Averages/Median
start_time = Time.now
test_array1 = [4.1, 5.6, 7.2, 1.7, 9.3, 4.4, 3.2]
test_array2 = [4.1, 7.2, 1.7, 9.3, 4.4, 3.2]

def find_median(array)
  array.sort!
  median = array[array.count/2] if array.count.odd?
  median = (array[array.count/2] + array[(array.count/2)-1])/2 if array.count.even?
  median
end

puts "The median of " + test_array1.inspect + " is: " + find_median(test_array1).to_s
puts "The median of " + test_array2.inspect + " is: " + find_median(test_array2).to_s
puts ((Time.now-start_time).to_f).to_s + "s"

File: median.swift
// http://rosettacode.org/wiki/Averages/Median

func median(array:[Float]) -> Float {
  var mutableArray = quicksort(array)
  if mutableArray.count % 2 == 1 {
    return mutableArray[mutableArray.count/2]
  }  else {
    return (mutableArray[mutableArray.count/2] + mutableArray[(mutableArray.count/2) - 1])/2
  }
}

func quicksort(array:[Float]) -> [Float] {
  if array.count <= 1 {
    return array
  }
  var lesser:[Float] = []
  var greater:[Float] = []
  var equiv:[Float] = []
  let pivot = array[array.count/2]
  for element in array {
    if element < pivot {
      lesser.append(element)
    } else if element > pivot {
      greater.append(element)
    } else {
      equiv.append(element)
    }
  }
  return quicksort(lesser) + equiv + quicksort(greater)
}

let testArray1:[Float] = [4.1, 5.6, 7.2, 1.7, 9.3, 4.4, 3.2]
let testArray2:[Float] = [4.1, 7.2, 1.7, 9.3, 4.4, 3.2]

println(median(testArray1))
println(median(testArray2))